frame/nomination-pools/src/lib.rs

```diff
@@ -1209,6 +1214,8 @@ pub mod pallet {
 		///
 		/// The removal can be voluntary (withdrawn all unbonded funds) or involuntary (kicked).
 		MemberRemoved { pool_id: PoolId, member: T::AccountId },
+		/// The roles of a pool have been updated to the given new roles.
+		RolesUpdated { root: T::AccountId, state_toggler: T::AccountId, nominator: T::AccountId },
```

Comment:
Just an idea: You could put the `ConfigOp` in here instead, then the UI would not need to know the prior state in case of a `ConfigOp::Set`.  
Otherwise the UI has to compare the old to the new version to find out if there was a change or noop.

Reply:
I think that's a bit superfluous, since now I have to also add some code to handle "what if root is set to `Some(old_root)`".

Basically the semantic of the event can be: 

1. It shows the diff
2. It shows the new state, regardless if there was a difference or not.

And I think the second option is a bit cleaner for the blockchain side. The UIs can read the old state if they want. 

Lastly, while I get that some UIs might only want to show the diff, I don't think that's the case at all times.